docs/07_how-to/14_ht-front-end-simple.mdx

```diff
@@ -0,0 +1,273 @@
+
+```
+export class NewPage extends FoundationElement {
+  myModal: Modal;
+}
+```
+
+In the **new-page.template.ts** add the element and a `ref` value so that the class can open it.
+
+In the click handler on the button, call the `show` method on the modal.
+
+```html
+...
+<rapid-button @click=${x => x.myModal.show()}></rapid-button>
+<rapid-modal ${ref('myModal')}>
+  <p>This paragraph will appear in the modal</p>
+</rapid-modal>
+...
+```
+
+When open, the modal is displayed over the top of any existing content in your page. It also creates an invisible overlay that prevents you user from interacting with any content outside the modal while it is active.
+
+## Adding a flyout
+To add a flyout to your application, add the `<rapid-flyout>` element to your template. 
+
+This element accepts a `closed` property value, a boolean for true or false. When the value is **true**, the flyout will be closed, and when the value is **false** the flyout will be open.
+
+By default, the flyout is positioned on the right side of the screen. To configure it to appear on the left, set `position="left"` on the element.
+
+You can insert any content you want within the flyout. 
+
+:::info
+When the user clicks outside the flyout, it closes. Also, you can add code to close the programmatically; set the value bound to the `closed` attribute.
+
+When a flyout is closed, the component emits a **closed** event, which you can listen to.
+:::
+
+Here is an example declaration of a flyout:
+
+```ts
+<rapid-button @click=${x => x.leftFlyoutActive = true}>Open left positioned flyout</rapid-button>
+<rapid-button @click=${x => x.rightFlyoutActive = true}>Open right positioned flyout</rapid-button>
+<rapid-flyout :closed="${x => !x.rightFlyoutActive}" @closed=${() => console.log('right flyout closed')}>
+  <p>This paragraph will appear in the flyout</p>
+</rapid-flyout>
+<rapid-flyout :closed="${x => !x.leftFlyoutActive}" position="left" @closed=${() => console.log('left flyout closed')}>
+  <p>This paragraph will appear in the flyout</p>
+</rapid-flyout>
+```
+
+## Example application
+We have created an [example application](https://github.com/genesislcap/platform-howto/tree/4a140c118916b177c0dd5fd355a5d5ec36f7d33b/howto-uiexamples) to demonstrate the things you can set up on the front end of a Genesis application.
+
+There are multiple UIs, with links to the corresponding code.
+
+### The back end
+Even if you are a front-end specialist, it is important to have some familiarity with the back end. You need to know what resources you can connect to in order to retrieve data and interact, and in many cases you will need to know exactly what fields are available on a specific resource.
+
+In our case, the back end is purposely kept simple, with little functionality.  It contains:
+
+- static data tables for Entity and Client
+- a Trade table with 10 pre-loaded trades
+- Kafka topic subscription for FX Rates, to show ticking data in the front end
```

Comment:
I can't see this in the howto-uiexamples. Kafka (pipelines) isn't out til 8.4 - what's the idea here? is it using that or something custom? if something custom do we have a plan to replace with pipelines when 8.4 comes out?